test/functionalTest/cases/4495_servicepath_levels_with_zero_length/4495_servicepath_levels_with_zero_length.test

```diff
@@ -0,0 +1,236 @@
+# Copyright 2024 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Servicapath with zero length levels are not allowed
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. Create entity E1/T1/A1=1, with servicepath '/Level_1' (success case)
+# 02. GET /v2/entities/E1, with servicepath '/Level_1' to see E1/T1/A1=1
+# 03. Create entity E2/T2/A2=2, with servicepath '/Level_1/Level_2' (success case)
+# 04. GET /v2/entities/E2, with servicepath '/Level_1/Level_2' to see E2/T2/A2=2
+# 05. Create entity E3/T3/A3=3, with servicepath '//Level_1' (SP level with 0 length), see error
+# 06. Create entity E3/T3/A3=3, with servicepath '///Level_1///Level_2//' (SP level with 0 length), see error
+# 07. Create entity E3/T3/A3=3, with servicepath '/Level_1///Level_2' (SP level with 0 length), see error
```

Comment:
Another interesting case to check could be:

```
# 0x. Create entity E3/T3/A3=3, with servicepath '/Level_1/Level_2//' (SP level with 0 length), see error
```

Could you include it, please?

Reply:
For above case ( i.e. servicepath = `/Level_1/Level_2//`), request will be executed successfully. Because before `servicePathCheck()` function call (Where the service paths are being checked), there is another function i.e. `removeTrailingSlash()` that is removing all the slash (`/`), present at the end of servicePath.
Please see below existing code snippet for the same:

![rest_servicePathSplit_function2](https://github.com/telefonicaid/fiware-orion/assets/55977298/ea65a182-e00c-4e5b-99fd-a7a4d22225f5)

![rest_removeTrailingSlash_function2](https://github.com/telefonicaid/fiware-orion/assets/55977298/25d8b769-854e-4128-86c3-4665250adca6)


See below API execution result:
![API_execution_result](https://github.com/telefonicaid/fiware-orion/assets/55977298/db723c31-d23d-4981-bec1-c34c420fe867)


I have applied some custom logs to see the value of servicePath, Please find below logs for reference:

![orion_API_execution_log](https://github.com/telefonicaid/fiware-orion/assets/55977298/2c70ecbe-7fca-4d6e-8add-9b41cf26e857)

So I think in that case, we don't need to include the above case in the FT. Please provide your opinion on this.

Reply:
Thank you very much for your detailed analysis!

It think this slash removal logic is inspired in web navigation (e.g. your browser doesn't distinguish between https://www.google.com/, https://www.google.com or https://www.google.com///////).

However, in the case of service paths I think is a bit weird removing *all slashes*.

I mean, removing *one slash* is ok, so `/Level1/Level2/` is equivalent to `/Level1/Level2`. However, if the user tries to use `/Level1/Level2//` probably he/she is doing something wrong that should be warned with an error response.

What do you think? Could the code be modified to implement this?

Reply:
Yes, I also think, It will be good to remove only 1 slash from servicePath, in case of multiple slash, an error message should be returned.

Reply:
updated code for the same.
Fixed in cc1e22a